File: frontend/src/app/shared/service/admin/alc/AdminAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {AdminDto} from 'src/app/shared/model/alc/Admin.model';
import {AdminCriteria} from 'src/app/shared/criteria/alc/AdminCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class AdminAdminService extends AbstractService<AdminDto, AdminCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): AdminDto {
        return new AdminDto();
    }

    public constrcutCriteria(): AdminCriteria {
        return new AdminCriteria();
    }

    get API() {
        return environment.apiUrlZynservice + 'admin/admin/';
    }
}

File: backend-ms1/src/main/java/ma/zs/zyn/service/facade/admin/alc/AdminAdminService.java
package ma.zs.zyn.service.facade.admin.alc;

import java.util.List;
import ma.zs.zyn.bean.core.alc.Admin;
import ma.zs.zyn.dao.criteria.core.alc.AdminCriteria;
import ma.zs.zyn.zynerator.service.IService;



public interface AdminAdminService extends  IService<Admin,AdminCriteria>  {
    Admin findByUsername(String username);
    boolean changePassword(String username, String newPassword);





}
